vignettes/Introduction_To_BiocParallel.Rmd

```diff
@@ -0,0 +1,867 @@
+categorization.](http://cran.r-project.org/web/views/HighPerformanceComputing.html)
+Most Task View packages cite or identify one or more of [*snow*](https://cran.r-project.org/package=snow) , [*Rmpi*](https://cran.r-project.org/package=Rmpi),[*multicore*](https://cran.r-project.org/package=multicore)  or [*foreach*](https://cran.r-project.org/package=foreach) as
+relevant parallelization infrastructure. Direct support in *R* for *parallel* computing started
+with release 2.14.0 with inclusion of the [parallel](https://cran.r-project.org/package=parallel) package which contains modified versions of
+[*multicore*](https://cran.r-project.org/package=multicore) and [*snow*](https://cran.r-project.org/package=snow) .
+
+
+A basic objective of [*BiocParallel*](http://bioconductor.org/packages/BiocParallel) is to reduce the complexity faced when developing
+and using software that performs parallel computations. With the
+introduction of the `BiocParallelParam` object, [*BiocParallel*](http://bioconductor.org/packages/BiocParallel) aims to provide a unified interface to
+existing parallel infrastructure where code can be easily executed in
+different environments. The `BiocParallelParam` specifies the environment of choice as well
+as computing resources and is invoked by 'registration' or passed as an
+argument to the [*BiocParallel*](http://bioconductor.org/packages/BiocParallel) functions.
+
+[*BiocParallel*](http://bioconductor.org/packages/BiocParallel) offers the following conveniences over the 'roll your own' approach to
+parallel programming.
+
+-   unified interface: `BiocParallelParam` instances define the method of parallel
+    evaluation (multi-core, snow cluster, etc.) and computing resources
+    (number of workers, error handling, cleanup, etc.).
+
+-   parallel iteration over lists, files and vectorized operations: `bplapply`, `bpmapply` and
+    `bpvec` provide parallel list iteration and vectorized operations. `bpiterate` iterates
+    through files distributing chunks to parallel workers.
+
+-   cluster scheduling: When the parallel environment is managed by a
+    cluster scheduler through [*batchtools](https://cran.r-project.org/package=batchtools), job management and result retrieval are
+    considerably simplified.
+
+-   support of `foreach` : The [*foreach*](https://cran.r-project.org/package=foreach) and [*iterators*](https://cran.r-project.org/package=iterators) packages are fully supported. Registration of
+    the parallel back end uses `BiocParallelParam` instances.
+
+Quick start
+===========
+
+The *BiocParallel* package is available at bioconductor.org and can be downloaded via `BiocManager`:
+
+```
+if (!requireNamespace("BiocManager", quietly = TRUE))
+     install.packages("BiocManager") 
+BiocManager::install("BiocParallel")
+```
+
+Load [*BiocParallel*](http://bioconductor.org/packages/BiocParallel)
+
+```
+{r}
+library(BiocParallel)
+
+```
```

Comment:
This code chunk (and others later) **must** have the `{r}` on the same line, without spaces, as the 'fence', so
~~~
```{r}
library(BiocParallel)
```
~~~

I'm guessing that this syntax is causing you problems when you try and 'knit' the vignette, because your R does not have the necessary packages installed. You have to install **all** the packages in the Depends:, Imports:, LinkingTo: and Suggests: fields of the DESCRIPTION field. One way of doing this is to use the following command
```
BiocManager::install("BiocParallel", dependencies = TRUE)
```
If you already have BiocParallel installed, then the command might be
```
BiocManager::install("BiocParallel", dependencies = TRUE, force = TRUE)
```

Reply:
Thank you very much @mtmorgan for this It helped me knit the vignette properly